backends/ref/ceed-ref-operator.c

```diff
@@ -769,76 +792,53 @@ static int CeedOperatorAssembleLinearDiagonal_Ref(CeedOperator op,
   ierr = CeedElemRestrictionGetNumElements(rstrin, &nelem); CeedChk(ierr);
   ierr = CeedBasisGetNumNodes(basisin, &nnodes); CeedChk(ierr);
   ierr = CeedBasisGetNumQuadraturePoints(basisin, &nqpts); CeedChk(ierr);
-  // Compute the diagonal of B^T D B
-  // Each node, qpt pair
-  CeedScalar *interpin, *interpout, *gradin, *gradout;
+  // Basis matrices
+  CeedScalar *identity = NULL, *interpin, *interpout, *gradin, *gradout;
+  bool evalNone = false;
+  for (CeedInt i=0; i<numemodein; i++)
+    evalNone = evalNone || (emodein[i] == CEED_EVAL_NONE);
+  for (CeedInt i=0; i<numemodeout; i++)
+    evalNone = evalNone || (emodeout[i] == CEED_EVAL_NONE);
+  if (evalNone) {
+    ierr = CeedCalloc(nqpts*nnodes, &identity); CeedChk(ierr);
+    for (CeedInt i=0; i<(nnodes<nqpts?nnodes:nqpts); i++)
+      identity[i*nnodes+i] = 1.0;
+  }
   ierr = CeedBasisGetInterp(basisin, &interpin); CeedChk(ierr);
   ierr = CeedBasisGetInterp(basisout, &interpout); CeedChk(ierr);
   ierr = CeedBasisGetGrad(basisin, &gradin); CeedChk(ierr);
   ierr = CeedBasisGetGrad(basisout, &gradout); CeedChk(ierr);
-  // Each element and component
-  CeedPragmaSIMD
```

Comment:
Did you mean to remove `CeedPragmaSIMD` in front of all loops of the interface?
There is also in `backends/ref/ceed-ref-restriction.c`, lines: 45-58.

Reply:
Yes, I didn't ensure single write access here, but I did in `backends/ref/ceed-ref-restriction.c`.